tests/e2e/utils.js

```diff
@@ -92,3 +125,140 @@ export const isBlockInsertedInWidgetsArea = async ( blockName ) => {
 				.length ) > 0
 	);
 };
+/**
+ * Visits the Site Editor main page in Core WordPress
+ *
+ * There are two different possible site editor pages:
+ *
+ * 1. `themes.php?page=gutenberg-edit-site` is the one used and available if the Gutenberg plugin is enabled.
+ * 2. `site-editor.php` is the one available in WP Core.
+ *
+ * @param {string} query String to be serialized as query portion of URL.
+ * @param {'core' | 'gutenberg'} [editorContext='core'] Whether to go to the Gutenberg URL or the Core one.
```

Comment:
I was curious and had a look at the util on gutenberg side and the query params seems to be used as JSON object but I've noticed we are using it as url `search` string. 

There is a wrong typing information on the [visitSiteEditor](https://github.com/WordPress/gutenberg/blob/a4bf9b68b81f78b477b9476021f2ddffeddabbfd/packages/e2e-test-utils/src/site-editor.js#L200-L213) util. if we passed query as a string to visitSiteEditor it would fail or at least the query part can produce unexpected results, no? The [addQueryArgs](https://github.com/WordPress/gutenberg/blob/trunk/packages/url/src/add-query-args.js) used underneath seems to expect an object of query params but then gets mutated into a string with `addQueryArgs`

[visitAdminPage](https://github.com/WordPress/gutenberg/blob/a4bf9b68b81f78b477b9476021f2ddffeddabbfd/packages/e2e-test-utils/src/visit-admin-page.js#L20) does indeed expect a string. I think we should handle the query to ensure it's compatible with both paths if we want to make both of them available.

Reply:
> There is a wrong typing information on the [visitSiteEditor](https://github.com/WordPress/gutenberg/blob/a4bf9b68b81f78b477b9476021f2ddffeddabbfd/packages/e2e-test-utils/src/site-editor.js#L200-L213) util. if we passed query as a string to visitSiteEditor it would fail or at least the query part can produce unexpected results, no? The [addQueryArgs](https://github.com/WordPress/gutenberg/blob/trunk/packages/url/src/add-query-args.js) used underneath seems to expect an object of query params but then gets mutated into a string with addQueryArgs

Good catch! I must have typed it early when I didn't use the query contructor util and then forgot to change it. The woes of not having TS!

> [visitAdminPage](https://github.com/WordPress/gutenberg/blob/a4bf9b68b81f78b477b9476021f2ddffeddabbfd/packages/e2e-test-utils/src/visit-admin-page.js#L20) does indeed expect a string. I think we should handle the query to ensure it's compatible with both paths if we want to make both of them available.

> I was curious and had a look at the util on gutenberg side and the query params seems to be used as JSON object but I've noticed we are using it as url search string.

Not sure I understand what you mean in those two, could you elaborate?